From b8d87f33755309ed770fe09d5ddafa3cf04da453 Mon Sep 17 00:00:00 2001
From: Rafael Martins <rmartins@anaconda.com>
Date: Tue, 16 Jul 2024 23:11:13 +0200
Subject: [PATCH 3/6] *: allow building with non-apple clang version.

we use upstream clang, that is not part of xcode, we need this patch to force building
with our clang.
---
 configure.cmake       | 4 ++--
 src/gn/CMakeLists.txt | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.cmake b/configure.cmake
index 79412cff8..3cbc96cf4 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -580,7 +580,7 @@ add_check_for_support(
    CONDITION MSVC OR
        (LINUX AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR
        (LINUX AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR
-       (MACOS AND CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+       (MACOS AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
    MESSAGE
        "${CMAKE_CXX_COMPILER_ID} compiler is not supported."
 )
@@ -590,7 +590,7 @@ add_check_for_support(
    CONDITION MSVC OR
        (LINUX AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR
        (LINUX AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR
-       (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang") OR
+       (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR
        (ANDROID AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR
        (MINGW AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR
        (MINGW AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
diff --git a/src/gn/CMakeLists.txt b/src/gn/CMakeLists.txt
index 0fe3e4e05..78c291b37 100644
--- a/src/gn/CMakeLists.txt
+++ b/src/gn/CMakeLists.txt
@@ -49,7 +49,7 @@ elseif(MSVC)
     set(platform msvc)
 elseif(MINGW)
     set(platform mingw)
-elseif(APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL AppleClang)
+elseif(APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL Clang)
     set(platform darwin)
 else()
     message(FATAL_ERROR "Unsupported gn platform !")
-- 
2.39.2 (Apple Git-143)

